components/script/dom/underlyingsourcecontainer.rs

```diff
@@ -111,6 +112,51 @@ impl UnderlyingSourceContainer {
         None
     }
 
+    /// <https://streams.spec.whatwg.org/#dom-underlyingsource-start>
+    /// The algorithm can return any value, including a promise,
+    /// we always transform the result into a promise for convenience.
```

Comment:
We could say it's because the spec specifically asks for it in [step 10](https://streams.spec.whatwg.org/#set-up-readable-stream-default-controller).

Reply:
Right.